scipp-imaging-2D.ipynb

```diff
@@ -9,433 +9,902 @@
+    "                                                   workspace_index=ws_index,\n",
+    "                                                   start_x=x_min_fit, end_x=x_max_fit)\n",
+    "                elastic_params = dict(zip(fit_elastic[\"parameters\"].value[\"Name\"].values, fit_elastic[\"parameters\"].value[\"Value\"].values))\n",
+    "                d_sample_elastic = elastic_params[\"f1.x0\"] / 2.0 # See fit table definition for extract x0\n",
+    "                lattice_strain = d_sample_elastic - d_sample\n",
     "\n",
-    "e = np.transpose(stitched[\"normalized_wl\"].variances)\n",
+    "                # define successful fitting\n",
+    "                success = (sample_fit.attrs[\"status\"].value == \"success\") and (fit_elastic.attrs[\"status\"].value == \"success\")\n",
     "\n",
-    "ws = sc.compat.mantid.to_workspace_2d(x, y, e, x_dim, instrument_file)\n",
-    "ws"
+    "                # fitted values STORED in list\n",
+    "                fit_list[i].append((ws_index, (row, col), bragg_edge, d_sample, d_sample_elastic, lattice_strain, success))\n",
+    "\n",
+    "                # workspace created from sample fit, workspace index 1 for fitted spectrum (index 0 for data, index 2 for difference curve)\n",
+    "                fitted_sample = sample_fit[\"workspace\"].data.values[\"empty\", 1]\n",
+    "                \n",
+    "                if sample_fit.attrs[\"status\"].value == \"success\":\n",
+    "                    spectrum_list_fitted_sample[i].append((fitted_sample.coords[\"wavelength\"].values, fitted_sample.values))\n",
+    "                else:\n",
+    "                    spectrum_list_fitted_sample[i].append((fitted_sample.coords[\"wavelength\"].values,\n",
+    "                                                           np.zeros_like(fitted_sample.values)))\n",
+    "                    \n",
+    "                fitted_elastic = fit_elastic[\"workspace\"].data.values[\"empty\", 1]\n",
+    "\n",
+    "                # workspace created from fit of sample under elastic deformation\n",
+    "                if fit_elastic.attrs[\"status\"].value == \"success\":\n",
+    "                    spectrum_list_fitted_sample_elastic[i].append((fitted_elastic.coords[\"wavelength\"].values, fitted_elastic.values))\n",
+    "                else:\n",
+    "                    spectrum_list_fitted_sample_elastic[i].append((fitted_elastic.coords[\"wavelength\"].values,\n",
+    "                                                                                               np.zeros_like(fitted_elastic.values)))\n",
+    "                plots_counter+=1\n",
+    "\n",
+    "    return fit_list, spectrum_list_fitted_sample, spectrum_list_fitted_sample_elastic\n",
+    "\n",
+    "fit_list, spectrum_list_fitted_sample, spectrum_list_fitted_sample_elastic = \\\n",
+    "    Bragg_edge_position(xpos_Bragg_edge=Bragg_edges_FCC,\n",
```

Comment:
Fitting step is also quite slow. Maybe @SimonHeybrock changes will address problems here, or more to the point try again after that refactoring completed.